src/modules/endpoint/endpoints/events.js

```diff
@@ -10,21 +10,24 @@ export default new EndpointBuilder('events')
     `/${state.router.params.language}/wp-json/extensions/v0/modified_content/events?since=1970-01-01T00:00:00Z`)
   .withMapper(json => json
     .filter(event => event.status === 'publish')
-    .map(event => (new EventModel({
-      id: event.id,
-      title: event.title,
-      content: event.content,
-      thumbnail: event.thumbnail,
-      address: event.location.address,
-      town: event.location.town,
-      startDate: moment(`${event.event.start_date} ${event.event.start_time}`),
-      endDate: moment(`${event.event.end_date} ${event.event.end_time}`),
-      allDay: event.event.all_day !== '0',
-      excerpt: event.excerpt,
-      availableLanguages: event.available_languages
-    })))
+    .map(event => {
+      const allDay = event.event.all_day !== '0'
+      return new EventModel({
+        id: event.id,
+        title: event.title,
+        content: event.content,
+        thumbnail: event.thumbnail,
+        address: event.location.address,
+        town: event.location.town,
+        startDate: moment(`${event.event.start_date} ${allDay ? '00:00:00' : event.event.start_time}`),
+        endDate: moment(`${event.event.end_date} ${allDay ? '23:59:59' : event.event.end_time}`),
+        allDay: allDay,
+        excerpt: event.excerpt,
+        availableLanguages: event.available_languages
+      })
+    })
     .filter(event => event.startDate.isValid())
-    .filter(event => event.startDate.isAfter(moment()))
+    .filter(event => event.startDate.isAfter(moment()) || (event.endDate.isValid() && event.endDate.isAfter(moment())))
```

Comment:
Do we need to filter for event.startDate.isAfter(moment()) and event.endDate.isAfter(moment())? I think filtering for event.endDate should be enough (+ startDate.isValid()), since the endDate should always be after startDate.

Reply:
If the endDate is not valid (e.g. because the backend serves an empty string for it), this implementation is still safe...